### PR TITLE
Feature/chkv v1

### PR DESCRIPTION
**Pull request by albertmink**
Ref to project resource https://sapjira.wdf.sap.corp/browse/ADTVEGA-672

### CODE REVIEW

File: file-formats/chkv/examples/z_aff_example_chkv.chkv.json [hunk 1]
```diff
@@ -63,8 +61,7 @@
             {
               "sign": "exclude",
               "option": "equals",
-              "low": "P",
-              "high": ""
+              "low": "P"
```
Comment (albertmink) [inline file-formats/chkv/examples/z_aff_example_chkv.chkv.json:64]: @bulcu is it ok to drop the empty `high`?

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      "! <p class="shorttext">Check Name</p>
+      "! The name of check
+      check_name TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Check Parameters</p>
+      "! Parameters of check
+      parameters TYPE ty_properties_tab,
+    END OF ty_check_parameter_json.
+
+  TYPES:
+       "! <p class="shorttext">Selected Checks</p>
```
Comment (albertmink) [inline file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:119]:
```suggestion
      "! <p class="shorttext">Selected Checks</p>
```

File: file-formats/chkv/chkv.json [hunk 3]
```diff
@@ -147,8 +151,7 @@
                           "Equals",
                           "Between",
                           "Greater than",
-                          "Less equal",
-                          "Contains pattern",
+                          "Contains patter",
```
Comment (wurzka) [inline file-formats/chkv/chkv.json:154]:
```suggestion
                          "Contains pattern",
```

Comment (wurzka) [reply file-formats/chkv/chkv.json:154]: This fits better with "not contains pattern"

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+      "! Equals
+      equals               TYPE ty_option VALUE 'EQ',
+      "! <p class="shorttext">Between</p>
+      "! Between
+      between              TYPE ty_option VALUE 'BT',
+      "! <p class="shorttext">Greater Than</p>
+      "! Greater than
+      greater_than         TYPE ty_option VALUE 'GT',
+      "! <p class="shorttext">Contains Pattern</p>
+      "! Contains patter
```
Comment (wurzka) [inline file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:23]: See above

**Review by wurzka — COMMENTED** [review]
One description issue, but also ask @schneidermic0 for review

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap (hunk 4 above)
Comment (albertmink) [inline file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:23]:
```suggestion
      "! Contains pattern
```

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+      "! $required
+      schema             TYPE string,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header             TYPE zif_aff_types_v1=>ty_header_60,
+      "! <p class="shorttext">Remote-Enabled Flag</p>
+      "! The ATC check variant is remote-enabled
+      remote_enabled     TYPE abap_bool,
+      "! <p class="shorttext">Referenced SCI Variant</p>
```
Comment (schneidermic0) [inline file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:139]:
```suggestion
      "! <p class="shorttext">Referenced Check Variant</p>
```

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    BEGIN OF ty_main,
+      "! <p class="shorttext">Schema</p>
+      "! Format version
+      "! $required
+      schema             TYPE string,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header             TYPE zif_aff_types_v1=>ty_header_60,
+      "! <p class="shorttext">Remote-Enabled Flag</p>
```
Comment (schneidermic0) [inline file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:136]: Minor: I wouldn't add 'Flag'

```suggestion
      "! <p class="shorttext">Remote-Enabled</p>
```

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    BEGIN OF ty_range,
+      "! <p class="shorttext">Sign</p>
+      "! The sign type of the range
+      sign   TYPE ty_sign,
+      "! <p class="shorttext">Option</p>
+      "! Sign option of the range
+      option TYPE ty_option,
+      "! <p class="shorttext">Low</p>
+      "! The low value of the range
+      low    TYPE string,
```
Comment (schneidermic0) [inline file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:68]: Shouldn't we add `$required` for `sign`, `Option` and `low`?

```suggestion
      "! <p class="shorttext">Sign</p>
      "! The sign type of the range
      "! $required
      sign   TYPE ty_sign,
      "! <p class="shorttext">Option</p>
      "! Sign option of the range
      "! $required
      option TYPE ty_option,
      "! <p class="shorttext">Low</p>
      "! The low value of the range
      "! $required
      low    TYPE string,
```

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+      "! Exclude
+      exclude TYPE ty_sign VALUE 'E',
+      "! <p class="shorttext">Include</p>
+      "! Include
+      include TYPE ty_sign VALUE 'I',
+    END OF co_sign.
+
+  TYPES:
+    "! <p class="shorttext">A Range</p>
+    "! A range of ATC check
```
Comment (schneidermic0) [inline file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:58]:
```suggestion
    "! <p class="shorttext">Range value</p>
    "! A range value for an ATC check parameter
```

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+    "! A parameter of ATC check
+    BEGIN OF ty_properties,
+      "! <p class="shorttext">Parameter Name</p>
+      "! The parameter name
+      "! $required
+      name             TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Parameter Type</p>
+      "! The parameter type
+      "! $required
+      type             TYPE string,
```
Comment (schneidermic0) [inline file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:90]: Can't we remove the field `type`

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+
+    ty_ranges      TYPE STANDARD TABLE OF ty_range WITH DEFAULT KEY,
+
+    "! <p class="shorttext">Entry in Value List</p>
+    "! A parameter value in the parameter value list
+    ty_param_value TYPE string,
+
+    "! <p class="shorttext">Parameter</p>
+    "! A parameter of ATC check
+    BEGIN OF ty_properties,
```
Comment (schneidermic0) [inline file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:82]: Shouldn't we use a more specific type name for the parameter?
```suggestion
    "! Parameter of the ATC check
    BEGIN OF ty_parameter,
```

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+      "! List of values for a multi-value check parameter
+      value_list       TYPE STANDARD TABLE OF ty_param_value WITH DEFAULT KEY,
+      "! <p class="shorttext">List of Range Values</p>
+      "! List of range values for a check parameter
+      value_range_list TYPE ty_ranges,
+    END OF ty_properties,
+
+    "! <p class="shorttext">Parameters</p>
+    "! List of check parameters
+    ty_properties_tab TYPE STANDARD TABLE OF ty_properties WITH DEFAULT KEY.
```
Comment (schneidermic0) [inline file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:104]:
```suggestion
    ty_parameters TYPE STANDARD TABLE OF ty_parameter WITH DEFAULT KEY.
```

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+
+  TYPES:
+    "! <p class="shorttext">Check</p>
+    "! Check with its parameters
+    BEGIN OF ty_check_parameter_json,
```
Comment (schneidermic0) [inline file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:109]: I would also suggest to rename the type to something like just `ty_check`

File: file-formats/chkv/format.md [hunk 13]
```diff
@@ -1,6 +1,6 @@
 # File Format for Object Type CHKV
 
-The file representation for object `Z_AFF_EXAMPLE_CHKV` is provided [here](./examples/z_aff_example_chkv.chkv.json).
+The file representation for an ATC check variant `Z_AFF_EXAMPLE_CHKV` is provided [here](./examples/z_aff_example_chkv.chkv.json).
```
Comment (schneidermic0) [inline file-formats/chkv/format.md:3]:
```suggestion
The file representation for example ATC check variant `Z_AFF_EXAMPLE_CHKV` is provided [here](./examples/z_aff_example_chkv.chkv.json).
```

File: file-formats/chkv/examples/z_aff_example_chkv.chkv.json (hunk 1 above)
Comment (schneidermic0) [reply file-formats/chkv/examples/z_aff_example_chkv.chkv.json:64]: Good question

File: file-formats/chkv/examples/z_aff_example_chkv.chkv.json [hunk 14]
```diff
@@ -52,7 +50,7 @@
               "sign": "exclude",
               "option": "equals",
               "low": "0002470721",
-              "high": "0000000000"
+              "high": "0000000001"
```
Comment (schneidermic0) [inline file-formats/chkv/examples/z_aff_example_chkv.chkv.json:53]: Does it make sense to have a Lowe `high` value than `low`

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header             TYPE zif_aff_types_v1=>ty_header_60,
+      "! <p class="shorttext">Remote-Enabled Flag</p>
+      "! The ATC check variant is remote-enabled
+      remote_enabled     TYPE abap_bool,
+      "! <p class="shorttext">Referenced SCI Variant</p>
+      "! Name of the referenced ATC check variant
+      referenced_variant TYPE zif_aff_types_v1=>ty_object_name_30,
```
Comment (schneidermic0) [inline file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:141]: Is the referenced variant supported by CHKV?

**Review by schneidermic0 — CHANGES_REQUESTED** [review]
Thanks.
I have added some questions and suggestions. Maybe, you want to check them.

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap (hunk 7 above)
Comment (albertmink) [reply file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:68]: @bulcu can you confirm that we need the `sign`, `option` and `low`?

File: file-formats/chkv/chkv.json [hunk 16]
```diff
@@ -64,55 +67,57 @@
...
                 },
                 "valueRangeList": {
-                  "title": "List of range values",
+                  "title": "List of Range Values",
                   "description": "List of range values for a check parameter",
                   "type": "array",
                   "items": {
-                    "title": "A Range",
-                    "description": "A range of ATC check",
+                    "title": "Range value",
```
Comment (albertmink) [inline file-formats/chkv/chkv.json:119]:
```suggestion
                    "title": "Range Value",
```

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap (hunk 7 above)
Comment (albertmink) [reply file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:68]: added keyword required

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap (hunk 15 above)
Comment (bulcu) [reply file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:146]: currently not but we have already provided this in the abap type, so we can support it later in the editor

File: file-formats/chkv/examples/z_aff_example_chkv.chkv.json (hunk 14 above)
Comment (wurzka) [reply file-formats/chkv/examples/z_aff_example_chkv.chkv.json]: We left the "high" field empty

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap (hunk 15 above)
Comment (schneidermic0) [reply file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:146]: If we do not support it, I would remove it.

File: file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap (hunk 9 above)
Comment (schneidermic0) [reply file-formats/chkv/type/zif_atc_aff_chkv_v1.intf.abap:94]: Talked to @bulcu: It is still required in the editor. We try to get rid of it. But until then, we keep it.

**Review by schneidermic0 — CHANGES_REQUESTED** [review]

**Review by schneidermic0 — APPROVED** [review]
Looks good to me, now. Thanks

**Pull request merged**